python/lib/communication/dmod/communication/maas_request/ngen/abstract_nextgen_request.py

```diff
@@ -150,24 +162,41 @@ def __init__(self,
         hydrofabric_data_id : str
             A data identifier for the hydrofabric, for distinguishing between different hydrofabrics that cover the same
             set of catchments and nexuses (i.e., the same sets of catchment and nexus ids).
+        bmi_cfg_data_id : str
+            The BMI init config ``data_id`` index, for identifying the particular BMI init config datasets
+            applicable to this request.
         catchments : Optional[Union[Set[str], List[str]]]
             An optional collection of the catchment ids to narrow the geospatial domain, where the default of ``None``
             or an empty collection implies all catchments in the hydrofabric.
-        bmi_cfg_data_id : Optional[str]
-            The optioanl BMI init config ``data_id`` index, for identifying the particular BMI init config datasets
-            applicable to this request.
+        partition_cfg_data_id : Optional[str]
+            The optional ``data_id`` of an existing dataset containing a partitioning config that should be used in the
+            requested job.
+        formulations : Optional[PartialRealizationConfig]
```

Comment:
Is it worth creating a new class hierarchy to handle this instead of handling this switch case based on if you are provided a partial config or a concrete config? So a new type that takes in a `PartialRealizationConfig`. Some class then would use that information in that message to create a `AbstractNgenRequest` subtype.

Reply:
In some ways I think that would be better.  I worry/thought it would not fit together well in certain places:  for example, the logic that is run during the `AWAITING_DATA_CHECK` and `AWAITING_DATA` job stages.

Reply:
Right, I guess in the short term to get around the job semantics you could have a job that schedules another job. The first job really just determines data requirements for the second. Alternatively, it might be easier to roll this functionality into its own service like I mentioned above.

Reply:
> Right, I guess in the short term to get around the job semantics you could have a job that schedules another job. The first job really just determines data requirements for the second. Alternatively, it might be easier to roll this functionality into its own service like I mentioned above.

I am concerned that will obfuscate things for the user.  We want/need to be able to start their job and give them back a job id right away, which they can use later to figure out how to get their output (or get back to a status page).  

We could still do something similar, albeit more complicated (e.g., have the initial job _spawn_ a sub-job for deriving data), but that is a larger scope that probably needs it's own future issue(s).